src/Core/Application/RealTime/UseCase/FindService/FindService.php

```diff
@@ -84,8 +188,86 @@ public function __invoke(
     public function createResponse(
         Service $service,
         array $downtimes,
-        ?Acknowledgement $acknowledgement
+        ?Acknowledgement $acknowledgement,
+        Host $host
     ): FindServiceResponse {
-        return new FindServiceResponse();
+        $findServiceResponse = new FindServiceResponse(
+            $service->getId(),
+            $service->getHostId(),
+            $service->getDescription(),
+            $service->getStatus(),
+            $service->getIcon(),
+            $service->getServicegroups(),
+            $downtimes,
+            $acknowledgement,
+            $host
+        );
+
+        $findServiceResponse->isFlapping = $service->isFlapping();
+        $findServiceResponse->isAcknowledged = $service->isAcknowledged();
+        $findServiceResponse->isInDowntime = $service->isInDowntime();
+        $findServiceResponse->output = $service->getOutput();
+        $findServiceResponse->performanceData = $service->getPerformanceData();
+        $findServiceResponse->commandLine = $service->getCommandLine();
+        $findServiceResponse->notificationNumber = $service->getNotificationNumber();
+        $findServiceResponse->lastStatusChange = $service->getLastStatusChange();
+        $findServiceResponse->lastNotification = $service->getLastNotification();
+        $findServiceResponse->latency = $service->getLatency();
+        $findServiceResponse->executionTime = $service->getExecutionTime();
+        $findServiceResponse->statusChangePercentage = $service->getStatusChangePercentage();
+        $findServiceResponse->nextCheck = $service->getNextCheck();
+        $findServiceResponse->lastCheck = $service->getLastCheck();
+        $findServiceResponse->hasPassiveChecks = $service->hasPassiveChecks();
+        $findServiceResponse->hasActiveChecks = $service->hasActiveChecks();
+        $findServiceResponse->lastTimeOk = $service->getLastTimeOk();
+        $findServiceResponse->severityLevel = $service->getSeverityLevel();
+        $findServiceResponse->checkAttempts = $service->getCheckAttempts();
+        $findServiceResponse->maxCheckAttempts = $service->getMaxCheckAttempts();
+
+        return $findServiceResponse;
+    }
+
+    /**
+     * Obfuscate passwords in the commandline
+     *
+     * @param Service $service
+     * @return string|null
+     */
+    private function obfuscatePasswordInServiceCommandLine(Service $service): ?string
```

Comment:
Is this function could be useful for other usecases ?

Reply:
Not yet it uses legacy code it will be refacto later